pkg/distributor/ha_tracker.go

```diff
@@ -0,0 +1,243 @@
+package distributor
+
+import (
+	"context"
+	"fmt"
+	"os"
+	// "net/http"
+	"sync"
+	"time"
+
+	"github.com/weaveworks/common/mtime"
+
+	"github.com/cortexproject/cortex/pkg/ingester/client"
+	"github.com/cortexproject/cortex/pkg/kvstore"
+)
+
+// Treat these as consts.
+var haInstanceBytes = []byte("prom_ha_instance")
+var haClusterBytes = []byte("prom_ha_cluster")
```

Comment:
These probably need to be configurable (in the config struct), and although I don't want to encourage bikeshedding, I'd go with `cluster` and `replica` as default.